pkg/apis/org/v1/backuprestorecommon_types.go

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package v1
+
+const (
+	USERNAME_SECRET_KEY              = "username"
+	PASSWORD_SECRET_KEY              = "password"
+	RESTIC_REPO_PASSWORD_SECRET_KEY  = "repo-password"
+	SSH_PRIVATE_KEY_SECRET_KEY       = "ssh-privatekey"
+	AWS_ACCESS_KEY_ID_SECRET_KEY     = "awsAccessKeyId"
+	AWS_SECRET_ACCESS_KEY_SECRET_KEY = "awsSecretAccessKey"
+)
+
+// +k8s:openapi-gen=true
+// List of supported backup servers
+type BackupServers struct {
+	// Rest server within the cluster.
+	// The server and configuration are created by operator when AutoconfigureRestBackupServer is true.
+	Internal RestServerConfig `json:"internal,omitempty"`
+	// Sftp backup server configuration.
+	// Mandatory fields are: RepoPassword, Hostname, Repo, Username, SshKeySecretRef.
+	Sftp SftpServerConfing `json:"sftp,omitempty"`
+	// Rest backup server configuration.
+	// Mandatory fields are: RepoPassword, Hostname.
+	Rest RestServerConfig `json:"rest,omitempty"`
+	// Amazon S3 or alternatives.
+	// Mandatory fields are: RepoPassword, Repo, AWS key+id or secret with it.
+	AwsS3 AwsS3ServerConfig `json:"awss3,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// SFTP backup server configuration
+// Example: user@host://srv/repo
```

Comment:
`:port`

Reply:
No, `:` here is a delimiter between server address and repository path. In case if port is present, the delimiter is `/`. See [docs](https://restic.readthedocs.io/en/latest/030_preparing_a_new_repo.html#sftp).